src/Microsoft.Diagnostics.EventFlow.Outputs.HttpOutput/HttpOutput.cs

```diff
@@ -0,0 +1,163 @@
+
+            this.httpClient = new HttpClient();
+            this.configuration = configuration;
+
+            if (string.IsNullOrWhiteSpace(this.configuration.ServiceUri)) {
+                var errMsg = $"{nameof(HttpOutput)}: no ServiceUri configured";
+                healthReporter.ReportProblem(errMsg);
+                throw new Exception(errMsg);
+            }
+
+            string userName = configuration.BasicAuthenticationUserName;
+            string password = configuration.BasicAuthenticationUserPassword;
+            bool credentialsIncomplete = string.IsNullOrWhiteSpace(userName) ^ string.IsNullOrWhiteSpace(password);
+            if (credentialsIncomplete)
+            {
+                errorMessage = $"{nameof(configuration)}: for basic authentication to work both user name and password must be specified";
+                healthReporter.ReportWarning(errorMessage, EventFlowContextIdentifiers.Configuration);
+                userName = password = null;
+            }
+
+            if (!string.IsNullOrWhiteSpace(userName) && !string.IsNullOrWhiteSpace(password)) 
+            {
+                string httpAuthValue = Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes(string.Format("{0}:{1}", userName, password)));
+                this.httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", httpAuthValue);
+            }
+
+            switch (this.configuration.Format) 
+            {
+                case "text":
+                    if (string.IsNullOrWhiteSpace(this.configuration.HttpContentType))
+                    {
+                        this.configuration.HttpContentType = "text/plain";
+                    }
+                    break;
+
+                case "json":
+                case "json-lines":
+                    if (string.IsNullOrWhiteSpace(this.configuration.HttpContentType))
+                    {
+                        this.configuration.HttpContentType = "application/json";
+                    }
+                    break;
+
+                default:
+                    errorMessage = $"{nameof(configuration)}: unknown Format \"{nameof(this.configuration.Format)}\" specified";
+                    healthReporter.ReportWarning(errorMessage, EventFlowContextIdentifiers.Configuration);
+                    break;
+            }
+        }
+
+        public async Task SendEventsAsync(IReadOnlyCollection<EventData> events, long transmissionSequenceNumber, CancellationToken cancellationToken)
+        {
+            if (events == null || events.Count == 0)
+            {
+                return;
+            }
+
+            try
+            {
+                var payload = new StringBuilder("");
+
+                switch (configuration.Format) 
+                {
+                    case "text":
+                        foreach (EventData evt in events)
+                        {
+                            payload.AppendLine(evt.ToString());
```

Comment:
Have you tested this?

EventData does not override ToString() so I do not think it will give you what you expect. It is going to produce "EventData object" or something similar for every event, no matter what data it carries.

I suggest for starters just dropping the text format and retaining json + json-lines